plugin/trino-spooling-filesystem/src/main/java/io/trino/spooling/filesystem/FileSystemSpoolingManager.java

```diff
@@ -129,6 +131,22 @@ public void acknowledge(SpooledSegmentHandle handle)
         fileSystem.deleteFile(location((FileSystemSpooledSegmentHandle) handle));
     }
 
+    @Override
+    public Optional<DirectLocation> directLocation(SpooledSegmentHandle handle)
+    {
+        FileSystemSpooledSegmentHandle fileHandle = (FileSystemSpooledSegmentHandle) handle;
+        try {
+            return fileSystem
+                    .preSignedUri(location(fileHandle), remainingTtl(fileHandle.expirationTime()))
+                    .map(location -> SpooledLocation.directLocation(location.uri(), headers(fileHandle)));
+        }
+        catch (UnsupportedOperationException | IOException e) {
+            // Fallback to coordinator pass-through when
+            // pre-signed URI cannot be generated or isn't supported.
+            return Optional.empty();
+        }
```

Comment:
we should check upfront if that is supported. 
If we expect it is but we get an exception here we should probably rethrow. Or at least log.

Reply:
We can't check uprfront

Reply:
This does not sound like resolution :)

Why can't we. Maybe we should expect this to be config property then?

Reply:
It can fail in the runtime anyway (service down)

Reply:
I am not sure we should go with fallback then. It may be that coordinator is absolutly not suited for passing large amounts of data, or can incurr cost to user.
I think if we want fallbacking mechanism it should be configurable and opt-in

Reply:
we have a lot of toggles right now, maybe let's gather some feedback first prior to adding more complexity?

Reply:
Reworked after the discussion